src/mlpack/methods/ann/loss_functions/poisson_nll_loss.hpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file poisson_nll_loss.hpp
+ * @author Mrityunjay Tripathi
+ *
+ * Definition of the Poisson NegativeLogLikelihood class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_POISSON_NLL_LOSS_HPP
+#define MLPACK_METHODS_ANN_LAYER_POISSON_NLL_LOSS_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Implementation of the poisson negative log likelihood layer. The poisson 
+ * negative log likelihood layer expects input for each class. 
+ * The layer also expects a class index, in the range between 1 and the
+ * number of classes, as target when calling the Forward function.
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class PoissonNLLLoss
+{
+ public:
+  /**
+   * Create the PoissonNLL object.
+   * 
+   * @param log_input if true the loss is computed as exp(input) - target*input, if
+   *        false then the loss is input - target*log(input+eps)
+   * @param full boolean value that tell if the the full loss has to be calculated
+   *        i.e to add Stirling approximation term
+   * @param eps A small value to prevent log(0) when log_input = False
+   * @param reduction specifies which reduction to be applied to the output
+   *        i.e sum'|'mean'. The corresponding integer values are 0 and 1.
+   */
+  PoissonNLLLoss(const bool log_input = true, const bool full = false,
```

Comment:
Pedantic style issue, please use camel casing for all names, see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#naming-conventions for more details. Also can you put each parameter on it's own line.

Reply:
Because of the long name ```PoissonNegativeLogLikelihoodLoss```, I preferred to write ```PoissonNLLLoss```.
Will that be fine or should I change?

Reply:
That's fine, can you still check the argument/parameter style, and use camel casing for all names.

Reply:
Hmm... There are still lots of other errors. I am using vscode and it has stopped showing any inline errors.